### ISSUE TITLE
[iOS] [V5] firebase.messaging().onMessage() doesn't trigger on real device but works on Simulator.

### ISSUE THREAD

**Issue by Steffi3rd** [invertase/react-native-firebase#3345]
Hello guys,

I check every issue and I didn't find an answer about my problem.

## Issue

I'm trying to display Data Messages on my real device. 

Steps :
1 — Create Message from Firebase Console
2 — Submit

_(Token is retrieved, permissions are enabled on device)_

**3.1 — On Simulator, the Data Messages is showing
3.2 — On real device, nothing.**

Here is the code : 

```
useEffect(() => {
        firebase.messaging().onMessage(message => {
          alert('message');
          console.log('message', message);
        });
}, []);
```

— I don't use Android, only iOS for now.

— I tried V6 and V5, I'm back to V5 now and it's not working.

— I don't use Firebase Notifications yet. 

— Everything works except that. Notifications on background works, the notification is display well with good content.

What missing for Data Messages ?

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
{
  "name": "peperonnie",
  "version": "1.0.0",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native run-ios && node node_modules/react-native/local-cli/cli.js start",
    "start-clean": "react-native run-ios && node node_modules/react-native/local-cli/cli.js start --reset-cache",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@react-native-community/async-storage": "^1.2.1",
    "@react-native-community/blur": "^3.3.1",
    "@react-native-community/cameraroll": "^1.1.1",
    "@react-native-community/datetimepicker": "^2.1.0",
    "@react-native-community/push-notification-ios": "^1.0.3",
    "@svgr/cli": "^4.1.0",
    "axios": "^0.18.0",
    "babel-plugin-transform-remove-console": "^6.9.4",
    "base-64": "^0.1.0",
    "fast-deep-equal": "^2.0.1",
    "firebase": "^5.9.4",
    "formik": "^1.5.1",
    "lodash": "^4.17.11",
    "moment": "^2.24.0",
    "moment-range": "^4.0.2",
    "react": "16.9.0",
    "react-content-loader": "^4.3.4",
    "react-native": "0.61.5",
    "react-native-fast-image": "7.0.2",
    "react-native-firebase": "^5.6.0",
    "react-native-gesture-handler": "^1.5.2",
    "react-native-image-crop-picker": "^0.26.1",
    "react-native-image-pan-zoom": "^2.1.11",
    "react-native-image-resizer": "^1.0.1",
    "react-native-keyboard-aware-scroll-view": "^0.8.0",
    "react-native-lightbox": "^0.8.0",
    "react-native-material-textfield": "^0.12.0",
    "react-native-modal": "^11.0.2",
    "react-native-progress-bar-animated": "^1.0.6",
    "react-native-reanimated": "^1.4.0",
    "react-native-screens": "^2.0.0-alpha.22",
    "react-native-svg": "^9.3.5",
    "react-native-swipeable": "^0.6.0",
    "react-native-typography": "^1.4.0",
    "react-navigation": "^4.0.10",
    "react-navigation-stack": "^1.7.3",
    "react-navigation-tabs": "^2.6.2",
    "rn-fetch-blob": "^0.10.15",
    "styled-components": "^4.1.3",
    "yup": "^0.27.0"
  },
  "devDependencies": {
    "@babel/core": "^7.7.5",
    "@babel/runtime": "^7.7.6",
    "@react-native-community/eslint-config": "^0.0.5",
    "babel-jest": "^24.9.0",
    "babel-plugin-module-resolver": "^3.2.0",
    "eslint": "^6.7.2",
    "jest": "^24.9.0",
    "metro-react-native-babel-preset": "^0.57.0",
    "react-test-renderer": "16.9.0"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```
</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
platform :ios, '9.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

target 'Peperonnie' do
  # Pods for Peperonnie
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'

  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'

  pod 'Firebase/Messaging', '~> 6.20.0'
  
  target 'PeperonnieTests' do
    inherit! :search_paths
    # Pods for testing
  end

  use_native_modules!
end

target 'Peperonnie-tvOS' do
  # Pods for Peperonnie-tvOS

  target 'Peperonnie-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
info Fetching system and libraries information...
System:
    OS: macOS 10.15.2
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Memory: 75.67 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.18.1 - /usr/local/opt/node@10/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/opt/node@10/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
  IDEs:
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - 5.6.0
- **`Firebase` module(s) you're using that has the issue:**
  - Firebase/Messaging
- **Are you using `TypeScript`?**
  - Non
  
</p>
</details>

**Comment by Salakar** [comment]
See https://github.com/invertase/react-native-firebase/pull/3339 - please try the v6.4.0-rc2 release

**Comment by Steffi3rd** [comment]
@Salakar Hey, thanks for your reply. 

I just read this again and I think you mentioned "[FIX] When the application is in the foreground, notification messages were not being sent to the onMessage handler (via willPresentNotification). This is now fixed." ?

But I'm in V5 right now because I intended to use Firebase Notifications (which will not work on V6). 

So if I understand well, `onMessage` never worked on V5 before? It's only fixed on V6 ? 🤔

**Comment by Ehesp** [comment]
A lot has changed since v5, including our own understanding of Messaging & iOS/Objective-C. We managed to fix a load of issues in that PR, however if you take a look at the code (especially for iOS), there is a significant amount of changes, some of which are not possible due to the way v5 is setup.

**Comment by Salakar** [comment]
#3339 has been merged and is the final RC is up - 6.4.0-rc4 - I'd recommend trying it out if you can. Unfortunately we don't have the resource to back port this to v5 given the size of the change and that v5 is almost a year old.
